src/ingestion/utils.py

```diff
@@ -50,53 +51,37 @@ def num_tokens_from_string(string: str, encoding_name: str = 'cl100k_base') -> i
     return num_tokens
 
 
-def merge_chunks(docs: List[Document], threshold: int) -> List[Document]:
+def merge_chunks(docs: List[Document], min_tkn_num: int, max_tkn_num: int) -> List[Document]:
```

Comment:
Grande! Più lineare della precedente no?
Metto qui sotto una proposta secondo me un po' più leggibile, ma se non ti torna mantieni la tua :)

```python
def merge_chunks(docs: List[Document], min_tkn_num: int, max_tkn_num: int) -> List[Document]:
    """ 
    Merges consecutive document chunks based on matching metadata and token count thresholds.

    Args:
        docs (List[Document]): A list of Document objects where each object represents a chunk of text. Each document is expected to have:
            - 'metadata': The header information, used to group paragraphs and subparagraphs.
            - 'page_content': The textual content of the document.
        min_tkn_num (int): The minimum number of tokens a document should have before considering merging.
        max_tkn_num (int): The maximum number of tokens allowed after merging chunks.

    Returns:
        List[Document]: A list of Document objects where smaller chunks (below the threshold) with matching metadata have been merged with following documents.
    """
    merged_docs = []
    i = 0

    while i < len(docs):
        merged_docs.append(deepcopy(docs[i]))  # deepcopy is needed here to prevent changes to 'merged_docs' from affecting the original 'docs' list

        # Try to merge the current document with subsequent ones while conditions are met:
        # - metadata matches
        # - the total number of tokens after merging is below the max token threshold
        # - at least one of the two documents has fewer tokens than the min token threshold
        i += 1
        while i < len(docs):
            cur_n_tkn = num_tokens_from_string(merged_docs[-1].page_content)
            next_n_tkn = num_tokens_from_string(docs[i].page_content)

            cond_same_meta = merged_docs[-1].metadata == docs[i].metadata
            cond_min_tkn = (cur_n_tkn < min_tkn_num) or (next_n_tkn < min_tkn_num)
            cond_max_tkn = (cur_n_tkn + next_n_tkn) < max_tkn_num

            if cond_same_meta and cond_min_tkn and cond_max_tkn:
                # Merge doc i in the current sequence and go to the next doc
                merged_docs[-1].page_content += "\n" + docs[i].page_content
                i += 1
            else:
                # Treat doc i as first of a new merging sequence
                break 
                
    return merged_docs

```

Reply:
Grazie @prh86, mi torna! Ho aggiornato la pull request